delegator/src/function/function.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common'
+import { Prisma } from '@prisma/client'
+import { PrismaService } from '../prisma.service'
+import { FunctionDto } from './dto/function.dto'
+import Caver from 'caver-js'
+
+@Injectable()
+export class FunctionService {
+  caver: any
+
+  constructor(private prisma: PrismaService) {
+    this.caver = new Caver(process.env.PROVIDER_URL)
+  }
+
+  encryptFunctionName(functionName: string) {
+    return this.caver.klay.abi.encodeFunctionSignature(functionName)
+  }
+
+  async create(functionDto: FunctionDto) {
+    const data: Prisma.FunctionUncheckedCreateInput = {
+      name: functionDto.name,
+      encodedName: this.encryptFunctionName(functionDto.name),
+      contractId: BigInt(functionDto.contractId)
+    }
+    return await this.prisma.function.create({ data })
+  }
+
+  async findAll(params: {
+    skip?: number
+    take?: number
+    cursor?: Prisma.FunctionWhereUniqueInput
+    where?: Prisma.FunctionWhereInput
+    orderBy?: Prisma.FunctionOrderByWithRelationInput
+  }) {
+    const { skip, take, cursor, where, orderBy } = params
+    const functions = await this.prisma.function.findMany({
+      skip,
+      take,
+      cursor,
+      where,
+      orderBy,
+      select: {
+        id: true,
+        name: true,
+        encodedName: true,
+        contract: { select: { address: true } }
+      }
+    })
+    return functions.map((M) => {
```

Comment:
It would be good to define it as a auxiliary function outside of this file (maybe `function.utils.ts`), and then use it for output of `findOne` as well to get the same output formatting no matter whether we use `findAll` or `findOne`